### ISSUE TITLE
Error with Twilio SMS, but only for some numbers, HTTP response 400: Cannot destructure property 'httpHeaders' of 'connectionData' as it is undefined.

### ISSUE THREAD

**Issue by asheroto** [RocketChat/Rocket.Chat#33194]
### Description:

For some reason when using Rocket Chat with Twilio SMS, I am getting this error message. The odd part is, it works fine if I'm texting from a regular cell phone number, but when texting from certain numbers like Google Voice, it fails with the error as the body of the HTTP request:

```
{"success":false,"error":"Cannot destructure property 'httpHeaders' of 'connectionData' as it is undefined."}
```

I don't see anything mentioned in the RocketChat Reports log area.

Twilio shows http response code `400` with message `bad request`. Twilio error is `11200` which is [HTTP retrieval failure](https://www.twilio.com/docs/api/errors/11200).

Again, if I send a message from a regular cell phone number, it appears to work. This is not a Google Voice issue, because I can text back and forth between all other numbers. This doesn't appear to be a Twilio issue, because Twilio is reporting HTTP status code 400.

My firewall is not blocking it, because in watching the logs I do see the POST request hit my server.

---

**Update**: as mentioned below, SMS fails even from the same carrier. My number through Verizon works, but my dad's number through Verizon does not work. My friend uses T-Mobile and hers works as well.

### Steps to reproduce:

1. Text number from cell phone number to confirm working
2. Create a free Google Voice account (takes < 5 minutes) and send a test message to RocketChat SMS number

### Expected behavior:

SMS should appear

### Actual behavior:

SMS does not appear, HTTP status code 400 results

### Server Setup Information:

- Version of Rocket.Chat Server: 6.12.1
- License Type: free
- Number of Users: 1
- Operating System: Debian
- Deployment Method: Docker
- Number of Running Instances: 1
- DB Replicaset Oplog: -
- NodeJS Version: -
- MongoDB Version: -

### Client Setup Information

- Desktop App or Browser Version: Chrome latest as of 2024-09-24 (not client issue)
- Operating System: Windows 11 (not client issue)

### Additional context

-

### Relevant logs:

I am getting these logs, but not sure if it's related. When I send a text and it goes through, I see 2x new `DeprecationWarning` messages. When I send a text that results in 400, it does NOT create one of these entries. This may be yet a separate issue, not sure.

```
{"level":40,"time":"2024-09-02T06:06:48.640Z","pid":1,"hostname":"*****","name":"DeprecationWarning","section":"API","msg":"The parameter \"query\" in the endpoint \"/api/v1/integrations.list\" is deprecated and will be removed on version 7.0.0"} 
{"level":40,"time":"2024-09-02T06:20:29.543Z","pid":1,"hostname":"*****","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"} 
{"level":40,"time":"2024-09-02T06:20:29.547Z","pid":1,"hostname":"*****","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"} 
{"level":40,"time":"2024-09-02T06:32:49.470Z","pid":1,"hostname":"*****","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"} 
{"level":40,"time":"2024-09-02T06:32:49.475Z","pid":1,"hostname":"*****","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"} 
{"level":40,"time":"2024-09-02T06:35:51.210Z","pid":1,"hostname":"*****","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"} 
{"level":40,"time":"2024-09-02T06:35:51.213Z","pid":1,"hostname":"*****","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"} 
```

FYI I tried reading the collecting logs article linked in your issue template, but the URL appears to have been moved? Looks like the GitHub issue template needs to be updated.
https://rocket.chat/docs/contributing/reporting-issues#gathering-logs

![image](https://github.com/user-attachments/assets/082bb553-2b26-47e0-a031-8b1fe00a6865)

**Comment by reetp** [comment]
Going to ask about this.

Please be patient.

**Comment by asheroto** [comment]
Thanks. Any update on this? 😊

**Comment by btcpbordeaux** [comment]
@reetp incoming sms with twilio as service is only producing 400s for us since 8/20. so i am guessing this has been broken since the update around 8/17 ? What is the status on this? This is a pretty big broken feature to have not working for so long

**Comment by asheroto** [comment]
So it works from my cell phone, my friend's cell phone, but not my dad's cell phone. My dad and I both have Verizon and my friend on T-Mobile. Meaning two different cell phones both on Verizon had different results. Wow.

So there's some big issue going on and I'm quite surprised that it hasn't been fixed yet especially since there are corporate clients. This makes me wonder how many clients we have missed the conversation with.

We are going to have to migrate to another platform unfortunately.

**Comment by reetp** [comment]
You don't need to @ people thanks.

I've told the devs. It's up to them - I have no say in it as I am not an employee.

Rocket is a big organisation. Github is a tiny tip. There are competing resources between sales, product, marketing, and development, primarily orientated around paying customers (and currently mainly focused on building v7)

It's how Rocket makes money, stays in business, and allows them to open source their code.

As a result they'll get to these things in line with their other priorities.

If it's urgent, get a subscription and you get support.

Otherwise the code is published/released as open source.

That doesn't mean Rocket are obliged to fix it - a fundamental misunderstanding by many. Anyone can do it.

If I hear anything I'll post back.

If anyone can try and nail the issue down more it may help get it looked at.

**Comment by asheroto** [comment]
Thanks for the info. I appreciate your time.

Do corporate clients / subscriptions run on the same code base, just with licensing added in?

Is the cloud-based system hosted by RocketChat the same code base?

**Comment by btcpbordeaux** [comment]
Apologies! Was hoping you were an employee by the way you had responded 

Aware of the rocketchat business model just was in panic mode and reaching out to all channels available. Have submitted a service ticket but thought I might check the issues and github to see if others were having similar issues since roughly 8/20. 

I have found the cause of the issue (for me at least), values for allowed content types add been placed either by a coworker or from an update to rocketchat. They were the default values apparently for that input in the client (image/*,audio/*,video/*,text/*,application/pdf). Removing these resulted in incoming texts working again.

It does not sound like your issue, sadly, as mine was plaguing every sms line.

**Comment by asheroto** [comment]
Unfortunately removing that line did not fix the issue here. The default media types:

```
image/*,audio/*,video/*,text/*,application/pdf
```

So it should work with them, of course.

**Comment by btcpbordeaux** [comment]
I'm a liar, that did not fix it for me, sadly. It appeared to at first but I was only replaying failed requests in the twilio portal. That also appears only related to file uploads and not content-type. I had noticed after removing that the replay failed message in the twilio console worked and received back empty <Response> twiml from RC; however, initial request to incoming sms endpoint still fails with a vague 400 "Invalid request".

**Comment by btcpbordeaux** [comment]
I think it is the security patch that was done a month ago. Twilio is sending requests with department=mydepartmentid
and it doesnt look like that is being taken into account.

currently have

	`isRequestFromTwilio(signature: string, requestBody: object): boolean {
		const authToken = settings.get<string>('SMS_Twilio_authToken');
		const siteUrl = settings.get<string>('Site_Url');

		if (!authToken || !siteUrl) {
			SystemLogger.error(`(Twilio) -> URL or Twilio token not configured.`);
			return false;
		}

		const twilioUrl = siteUrl.endsWith('/')
			? `${siteUrl}api/v1/livechat/sms-incoming/twilio`
			: `${siteUrl}/api/v1/livechat/sms-incoming/twilio`;
		return twilio.validateRequest(authToken, signature, twilioUrl, requestBody);
	}

	validateRequest(request: Request): boolean {
		// We're not getting original twilio requests on CI :p
		if (process.env.TEST_MODE === 'true') {
			return true;
		}
		const twilioHeader = request.headers['x-twilio-signature'] || '';
		const twilioSignature = Array.isArray(twilioHeader) ? twilioHeader[0] : twilioHeader;
		return this.isRequestFromTwilio(twilioSignature, request.body);
	}`

maybe should change that to something like:

`isRequestFromTwilio(signature: string, request: Request): boolean {
	const authToken = settings.get<string>('SMS_Twilio_authToken');
        const siteUrl = settings.get<string>('Site_Url');

	if (!authToken) {
		SystemLogger.error(`(Twilio) -> Twilio auth token not configured.`);
		return false;
	}
        if (!request.url.startsWith(siteUrl)) return false;

	return twilio.validateRequest(authToken, signature, request.url, request.body);
}

validateRequest(request: Request): boolean {
		// We're not getting original twilio requests on CI :p
		if (process.env.TEST_MODE === 'true') {
			return true;
		}
		const twilioHeader = request.headers['x-twilio-signature'] || '';
		const twilioSignature = Array.isArray(twilioHeader) ? twilioHeader[0] : twilioHeader;
		return this.isRequestFromTwilio(twilioSignature, request);
	}`
	
	or whatever property has the actual url including querystring params. should just check against the siturl, but then need to use the url with the params if querystring params (like department are present)

**Comment by asheroto** [comment]
If you run a previous version does the issue go away for you?

**Comment by btcpbordeaux** [comment]
Yes, if I run before security patch, no issues. Also if I remove ?department=dept_id from the twilio webhook, it works fine

**Comment by btcpbordeaux** [comment]
mine is a separate issue from yours based on my error and what I seen in the source code, I am pretty sure

**Comment by asheroto** [comment]
On my side when logging inbound requests, I don't see it in the incoming requests when I start logging the connection. So yes, I think the issue is separate.

**Comment by asheroto** [comment]
I turned on debug logging and checked the logs via docker logs.

These are the logs when using the working example number:

```
{"level":20,"time":"2024-09-19T05:04:37.788Z","pid":1,"hostname":"***REDACTED***","name":"Livechat","msg":"New incoming conversation: id: undefined | token: ***REDACTED***"}
{"level":20,"time":"2024-09-19T05:04:37.790Z","pid":1,"hostname":"***REDACTED***","name":"Livechat","msg":"Found matching user by token"}
{"level":20,"time":"2024-09-19T05:04:37.793Z","pid":1,"hostname":"***REDACTED***","name":"Livechat","msg":"Attempting to find or create a room for visitor ***REDACTED***"}
{"level":20,"time":"2024-09-19T05:04:37.830Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id notifyUsersOnMessage for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.831Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id before-get-mentions-get-teams for hook beforeGetMentions"}
{"level":20,"time":"2024-09-19T05:04:37.841Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id after-omnichannel-save-message for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.842Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id livechat-resume-on-hold for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.847Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id save-last-visitor-message-timestamp for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.848Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id livechat-send-crm-message for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.848Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id save-visitor-inactivity for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.848Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id leadCapture for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.848Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id livechat-send-crm-lead-capture for hook livechat.leadCapture"}
{"level":20,"time":"2024-09-19T05:04:37.849Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id saveAnalyticsData for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.849Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id markRoomNotResponded for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.850Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id save-last-message-to-inquiry for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.854Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id sendMessageBySms for hook afterOmnichannelSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.858Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id message-read-receipt-afterSaveMessage for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.858Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id engagementDashboard.afterSaveMessage for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.859Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id sendNotificationsOnMessage for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.861Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id before-get-mentions-get-teams for hook beforeGetMentions"}
{"level":20,"time":"2024-09-19T05:04:37.865Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id emojione-shortnameToUnicode for hook beforeSendMessageNotifications"}
{"level":20,"time":"2024-09-19T05:04:37.874Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id filter-markdown for hook renderNotification"}
{"level":20,"time":"2024-09-19T05:04:37.875Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id mention-user-not-in-channel for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.875Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id API_Embed for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.876Z","pid":1,"hostname":"***REDACTED***","name":"OEmbed","msg":"Parsing message URLs"}
{"level":20,"time":"2024-09-19T05:04:37.876Z","pid":1,"hostname":"***REDACTED***","name":"OEmbed","msg":"URLs found 0"}
{"level":20,"time":"2024-09-19T05:04:37.877Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id ReplyEmail for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.877Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id integrations-sendMessage for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.878Z","pid":1,"hostname":"***REDACTED***","name":"Integrations","section":"Outgoing WebHook","msg":"Execute Trigger:","arg":"sendMessage"}
{"level":20,"time":"2024-09-19T05:04:37.878Z","pid":1,"hostname":"***REDACTED***","name":"Integrations","section":"Outgoing WebHook","msg":"Got the event arguments for the event: sendMessage","argObject":{"event":"sendMessage","message":{"_id":"***REDACTED***REDACTED***","rid":"***REDACTED***","token":"***REDACTED***","msg":"test","attachments":[],"alias":"***REDACTED***","ts":"2024-09-19T05:04:37.795Z","u":{"_id":"***REDACTED***REDACTED***","username":"***REDACTED***","name":"***REDACTED***"},"_updatedAt":"2024-09-19T05:04:37.825Z","urls":[],"mentions":[],"channels":[],"md":[{"type":"PARAGRAPH","value":[{"type":"PLAIN_TEXT","value":"test"}]}]},"room":{"room":{"_id":"***REDACTED***REDACTED***","cl":false,"estimatedWaitingTimeQueue":9999999,"fname":"***REDACTED***","lm":"2024-09-19T05:02:19.727Z","msgs":10,"open":true,"priorityWeight":99,"queuedAt":"2024-09-19T04:43:37.142Z","sms":{"from":"+***REDACTED***"},"source":{"type":"sms","alias":"twilio"},"t":"l","ts":"2024-09-19T04:43:37.142Z","usersCount":2,"v":{"_id":"***REDACTED***REDACTED***","username":"***REDACTED***","token":"***REDACTED***","status":"online","activity":["2024-07","2024-08","2024-09"],"lastMessageTs":"2024-09-19T05:02:19.727Z"},"_updatedAt":"2024-09-19T05:02:19.796Z","lastMessage":{"_id":"***REDACTED***","rid":"***REDACTED***","token":"***REDACTED***","msg":"hi","attachments":[],"alias":"***REDACTED***","ts":"2024-09-19T05:02:19.727Z","u":{"_id":"***REDACTED***REDACTED***","username":"***REDACTED***","name":"***REDACTED***"},"_updatedAt":"2024-09-19T05:02:19.753Z","urls":[],"mentions":[],"channels":[],"md":[{"type":"PARAGRAPH","value":[{"type":"PLAIN_TEXT","value":"hi"}]}]},"metrics":{"v":{"lq":"2024-09-19T05:00:59.435Z"},"reaction":{"fd":"2024-09-19T05:00:54.901Z","ft":2.288,"tt":2.288},"response":{"avg":1037.67,"fd":"2024-09-19T05:00:54.901Z","ft":1037.67,"total":1,"tt":1037.67},"servedBy":{"lr":"2024-09-19T05:00:54.847Z"}},"servedBy":{"_id":"***REDACTED***REDACTED***","username":"***REDACTED***","ts":"2024-09-19T05:00:52.613Z"},"waitingResponse":true},"roomUpdater":{"dirty":false,"_inc":{},"_set":{}}}}}
{"level":20,"time":"2024-09-19T05:04:37.878Z","pid":1,"hostname":"***REDACTED***","name":"Integrations","section":"Outgoing WebHook","msg":"Starting search for triggers for the room: undefined"}
{"level":20,"time":"2024-09-19T05:04:37.878Z","pid":1,"hostname":"***REDACTED***","name":"Integrations","section":"Outgoing WebHook","msg":"Found 0 to iterate over and see if the match the event."}
{"level":20,"time":"2024-09-19T05:04:37.879Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id PropagateDiscussionMetadata for hook afterSaveMessage"}
{"level":20,"time":"2024-09-19T05:04:37.879Z","pid":1,"hostname":"***REDACTED***","name":"Callbacks","msg":"Executing callback with id threads-after-save-message for hook afterSaveMessage"}
{"level":35,"time":"2024-09-19T05:04:37.883Z","pid":1,"hostname":"***REDACTED***","name":"API","method":"POST","url":"/api/v1/livechat/sms-incoming/twilio","userAgent":"TwilioProxy/1.1","length":"568","host":"chat.***REDACTED***","remoteIP":"***REDACTED***","responseTime":103}
{"level":35,"time":"2024-09-19T05:04:37.885Z","pid":1,"hostname":"***REDACTED***","name":"Meteor","method":"livechat:setCustomField","userId":null,"instanceId":"***REDACTED***"}
{"level":40,"time":"2024-09-19T05:04:37.885Z","pid":1,"hostname":"***REDACTED***","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"}
{"level":35,"time":"2024-09-19T05:04:37.887Z","pid":1,"hostname":"***REDACTED***","name":"Meteor","method":"livechat:setCustomField","userId":null,"instanceId":"***REDACTED***"}
{"level":40,"time":"2024-09-19T05:04:37.887Z","pid":1,"hostname":"***REDACTED***","name":"DeprecationWarning","section":"METHOD","msg":"The method \"livechat:setCustomField\" is deprecated and will be removed on version 7.0.0"}
{"level":35,"time":"2024-09-19T05:04:37.992Z","pid":1,"hostname":"***REDACTED***","name":"Meteor","method":"getRoomByTypeAndName","userId":"***REDACTED***","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Rocket.Chat/4.1.0 Chrome/124.0.6367.243 Electron/30.4.0 Safari/537.36","referer":"https://chat.***REDACTED***/live/***REDACTED***/room-info","remoteIP":"***REDACTED***","instanceId":"***REDACTED***"}
{"level":35,"time":"2024-09-19T05:04:37.997Z","pid":1,"hostname":"***REDACTED***","name":"API","method":"POST","url":"/api/v1/method.call/getRoomByTypeAndName","userId":"***REDACTED***","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Rocket.Chat/4.1.0 Chrome/124.0.6367.243 Electron/30.4.0 Safari/537.36","length":"125","host":"chat.***REDACTED***","referer":"https://chat.***REDACTED***/live/***REDACTED***/room-info","remoteIP":"***REDACTED***","status":200,"responseTime":6}
```

These are the logs when using the non-working example number:
```
{"level":35,"time":"2024-09-19T05:04:54.665Z","pid":1,"hostname":"***REDACTED***","name":"API","method":"POST","url":"/api/v1/subscriptions.read","userId":"***REDACTED***","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Rocket.Chat/4.1.0 Chrome/124.0.6367.243 Electron/30.4.0 Safari/537.36","length":"27","host":"chat.***REDACTED***","referer":"https://chat.***REDACTED***/live/***REDACTED***/room-info","remoteIP":"***REDACTED***","status":200,"responseTime":15}
{"level":20,"time":"2024-09-19T05:05:00.711Z","pid":1,"hostname":"***REDACTED***","name":"Livechat","msg":"New incoming conversation: id: undefined | token: ***REDACTED***"}
{"level":20,"time":"2024-09-19T05:05:00.714Z","pid":1,"hostname":"***REDACTED***","name":"Livechat","msg":"No matches found. Attempting to create new user with token ***REDACTED***"}
{"level":20,"time":"2024-09-19T05:05:00.714Z","pid":1,"hostname":"***REDACTED***","name":"Livechat","msg":"Saving connection data for visitor ***REDACTED***"}
{"level":35,"time":"2024-09-19T05:05:00.721Z","pid":1,"hostname":"***REDACTED***","name":"API","method":"POST","url":"/api/v1/livechat/sms-incoming/twilio","userAgent":"TwilioProxy/1.1","length":"581","host":"chat.***REDACTED***","remoteIP":"44.203.204.28","err":{"type":"TypeError","message":"Cannot destructure property 'httpHeaders' of 'connectionData' as it is undefined.","stack":"TypeError: Cannot destructure property 'httpHeaders' of 'connectionData' as it is undefined.\n    at app/livechat/server/lib/LivechatTyped.ts:658:13\n    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40\n => awaited here:\n    at Function.Promise.await (/app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:56:12)\n    at app/livechat/imports/server/rest/sms.ts:76:23\n    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40\n => awaited here:\n    at Function.Promise.await (/app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:56:12)\n    at app/livechat/imports/server/rest/sms.ts:119:16\n    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40\n => awaited here:\n    at Function.Promise.await (/app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:56:12)\n    at packages/meteor.js:367:22\n    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1389:31)\n    at app/api/server/api.ts:649:39\n    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40"},"status":400,"responseTime":15,"msg":"Cannot destructure property 'httpHeaders' of 'connectionData' as it is undefined."}
```

**Comment by reetp** [comment]
Not sure but this PR may affect this.

Please test 6.12.2 and let us know thanks.

https://github.com/RocketChat/Rocket.Chat/pull/33306

**Comment by asheroto** [comment]
I'm using RocketChat as a Docker container, so I'll test as soon as the next version is released.

For now, I set up a failover SMS through Twilio, so not time-sensitive but still important. 😊

**Comment by asheroto** [comment]
FYI I set up a SaaS and the phone numbers work correctly. The set up a trial, used the same SID and token, and changed the webhook URL to the SaaS URL and now can receive all text messages. This confirms that it is NOT a Twilio configuration issue, but rather an issue with the code.

**Comment by llvee** [comment]
@asheroto 

Have you considered network routing being the primary issue? Since you are only having issues with certain numbers that is a lot more likely than the code being at fault in this case. Testing a larger base of numbers would help better determine the exact cause.

**Comment by julio-cfa** [comment]
Hey @asheroto,

In theory, the code for self-hosted and cloud-hosted workspaces, in regards to the Twilio integration, is the same. It's indeed a very weird issue...

**Comment by llvee** [comment]
@asheroto  

It looks like my last communication yesterday either didn't get posted here or was removed. 

Your issue is likely occuring due to telephony routing. To confirm, you can test more numbers from different carriers. Besides that you should also verify that you have the same configuration for both of the Verizon numbers on the Twilio account, in order to verify that it's not a Twilio configuration issue which will also help create more proof for the former resolution hypothesis.

**Comment by asheroto** [comment]
@llvee  I appreciate your response, but just a bit confused on how it could be telephony routing, if the only thing I change is the webhook URL in Twilio, and it works with SaaS.

**Comment by reetp** [comment]
There should be no code differences between "self-hosted" and "SaaS" code of the same version.

You can check the commit details in admin/Workspace to make sure they are identical.

Please post both here for comparison.

**Comment by asheroto** [comment]
The admin/Workspace area shows `6.12.1`:
![image](https://github.com/user-attachments/assets/4b441c02-0ecd-4d6f-90e6-04d67fa03d28)

Although using Docker I set the `RC_VERSION` and `RELEASE` vars to `6.13.0-rc.3` when starting. I couldn't test `6.12.2` because it wasn't available as a release. I realize I could build it but am just holding out for the sake of time.

Since I'm using Cloudflare Tunnel, I thought, maybe it's somehow blocking or malforming certain information, but then I set it up directly to my instance and had the same issue.

I haven't modified any files within the Docker volume, so hoping a new release fixes this? Such a weird issue.

**Comment by asheroto** [comment]
I did see the invite. I've been waiting for another release to see if it potentially resolved the issue.

Since when changing the webhook URL to the SaaS URL the issue goes away, I'm not convinced it's a telephony issue.

I have a failover webhook set up so the issue is not critical at this point just important. Once the next release comes out I'll try that.

**Comment by asheroto** [comment]
I'm happy to report that version `6.13.0` appears to have resolved this problem. I texted from Google Voice and two other VoIP application, as well as another cell phone, and the system did not generate an error and use the fallback webhook. Thanks for your time everyone! 😊

**Comment by reetp** [comment]
>  if you would like more help with anything technology related you're welcome to chat or communicate further with me as time allows

Please note that this is an issue tracker, bot a room for touting for business.

> It seems that there is likely a telephony routing issue with the other version

This is incorrect.

Likely a duplicate or closely related to this:

https://github.com/RocketChat/Rocket.Chat/issues/33318

As per the OP this has been fixed in the latest 6.13.x release and I think it has been backported as well.
